src/main/java/org/mifos/processor/bulk/camel/routes/ProcessorStartRoute.java

```diff
@@ -233,19 +267,158 @@ private void setup() {
+                                "The file uploaded is too big. " +
+                                        "Please upload a file and try again.");
+                    }
+                    else if(!verifyCsv(file)){
+                        setErrorResponse(exchange, 400, "Invalid file structure",
+                                "The file uploaded contains wrong structure." +
+                                        " Please upload correct file columns and try again.");
+                    }
+                    else {
+                        logger.info("\n\n Filename: " + fileName + " \n\n");
+                        setResponse(exchange, 200);
+                    }
+
+                })
+                .log("Completed route direct:validateFileSyncResponse");
+
+        from("direct:validate-file")
+                .id("direct:validate-file")
+                .log("Starting route direct:validate-file")
+                .process(exchange -> {
+                    System.setProperty("mail.mime.multipart.ignoreexistingboundaryparameter", "true");
+                    System.setProperty("mail.mime.multipart.allowempty", "true");
+                    InputStream inputStreams = exchange.getIn().getBody(InputStream.class);
+                    MimeBodyPart mimeMessage = new MimeBodyPart(inputStreams);
+                    DataHandler dh = mimeMessage.getDataHandler();
+                    logger.info("File name: " + dh.getName());
+                    Tika tika = new Tika();
+                    String fileType = tika.detect(dh.getName());
+                    logger.info("File type: " + fileType);
+                    if (dh.getName() == null || dh.getName().isEmpty() ) {
+                        setErrorResponse(exchange, 400, "File not uploaded",
+                    "There was no fie uploaded with the request. " +
+                            "Please upload a file and try again.");
+                    }
+                    else if (!fileType.equalsIgnoreCase("text/csv")) {
+                        setErrorResponse(exchange, 400, "Broken file",
+                    "The file uploaded is broken as it has a different extension. " +
+                            "Please upload a csv file and try again.");
+                    }
+                 else{
+                     setResponse(exchange, 200);}
+
+    });}
+
+    private boolean verifyData(File file) throws IOException {
+        InputStream ips = new FileInputStream(file);
+        InputStreamReader ipsr = new InputStreamReader(ips);
+        BufferedReader br = new BufferedReader(ipsr);
+        String line;
+        br.readLine();
+        while ((line = br.readLine()) != null) {
+            String[] row = line.split(",");
+            logger.info("row is" + Arrays.toString(row));
+            if (row.length != columnNames.size()) {
+                logger.info("Row invalid " + row.length + " " + columnNames.size());
```

Comment:
we have to provide more informative error messages,  please add a ticket to the back log to aggregate these error messages